terraform/environments/sprinkler/ssm.tf

```diff
@@ -0,0 +1,163 @@
+		"Values": ["yes", "true"]
+	}]
+}
+JSON
+  }
+}
+
+###### Approval rule #####
+
+resource "aws_ssm_patch_baseline" "patch-baseline-poc" {
+  name             = "${local.application_name}-baseline"
+  operating_system = local.operating_system
+
+  approval_rule {
+    approve_after_days = 7
+    compliance_level   = "HIGH"
+
+    patch_filter {
+      key    = "CLASSIFICATION"
+      values = ["Security", "Bugfix"]
+    }
+  }
+}
+
+###### ssm maintenance window #####
+
+
+resource "aws_ssm_maintenance_window" "ssm-maintenance-window" {
+  name     = "${local.application_name}-maintenance-window"
+  schedule = "cron(0 15 ? * WED *)"
+  duration = 3
+  cutoff   = 1
+}
+
+
+###### ssm maintenance target #####
+
+resource "aws_ssm_maintenance_window_target" "ssm-targets" {
+  window_id     = aws_ssm_maintenance_window.ssm-maintenance-window.id
+  name          = "maintenance-window-target"
+  description   = "This is a maintenance window target"
+  resource_type = "INSTANCE"
+
+  targets {
+    key    = "tag:Patching"
+    values = ["yes"]
+  }
+}
+
+
+
+
+###### ssm maintenance task #####
+resource "aws_ssm_maintenance_window_task" "ssm-maintenance-window-command-task" {
+  name            = "${local.application_name}-command-patching-task"
+  max_concurrency = 2
+  max_errors      = 1
+  priority        = 1
+  task_arn        = "AWS-RunPatchBaseline"
+  task_type       = "RUN_COMMAND"
+  window_id       = aws_ssm_maintenance_window.ssm-maintenance-window.id
+
+
+  targets {
+    key    = "WindowTargetIds"
+    values = aws_ssm_maintenance_window_target.ssm-targets.*.id
+  }
+
+  task_invocation_parameters {
+    run_command_parameters {
+      output_s3_bucket = "${local.application_name}-patching-logs"
+      timeout_seconds  = 6000
+      service_role_arn = aws_iam_role.ssm_ec2_instance_role.arn
+
+      # Install will perform a scan followed by an install on any missing packages. This is completed with a system
+      # rebot. NOTE: Scan only can be used to output results to S3 log bucket and does not cause a rebot.
+      parameter {
+        name   = "Operation"
+        values = ["Scan"]
+      }
+    }
+  }
+}
+
+
+resource "aws_ssm_maintenance_window_task" "ssm-maintenance-window-automation-task" {
+  name            = "${local.application_name}-automation-patching-task"
+  max_concurrency = 2
+  max_errors      = 1
+  priority        = 1
+  task_type       = "AUTOMATION"
+  task_arn        = "AWS-PatchAsgInstance"
+  window_id       = aws_ssm_maintenance_window.ssm-maintenance-window.id
+
+
+  targets {
+    key    = "WindowTargetIds"
+    values = aws_ssm_maintenance_window_target.ssm-targets.*.id
+  }
+
+  task_invocation_parameters {
+    automation_parameters {
+      document_version = "$LATEST"
+
+      parameter {
+        name   = "InstanceId"
+        values = aws_ssm_maintenance_window_target.ssm-targets.*.id
+      }
+    }
+  }
+}
+
+###### s3 Bucket Patch Logs #####
+
+resource "aws_s3_bucket" "patching-log-bucket" {
```

Comment:
As part of further refinement, would be good to use the s3 module for this bucket.